### PR TITLE
KAFKA-17406: Move ClientIdAndBroker to server-common module

### PR DESCRIPTION
**Pull request by wernerdv**
Moved ClientIdAndBroker to server-common module ([JIRA](https://issues.apache.org/jira/browse/KAFKA-17406)).

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by mumrah — CHANGES_REQUESTED** [review]
Thanks for the patch @wernerdv! 

I think server-common is a better place for this. As you saw, the only usage is the AbstractFetcherThread which really has nothing to do with the metadata module.

**Comment by wernerdv** [comment]
@mumrah PTAL

**Comment by wernerdv** [comment]
@chia7712 Hello, please review PR.

File: server-common/src/main/java/org/apache/kafka/common/ClientIdAndBroker.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        this.brokerPort = brokerPort;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%s-%s-%d", clientId, brokerHost, brokerPort);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (mimaison) [inline server-common/src/main/java/org/apache/kafka/common/ClientIdAndBroker.java]: Do we actually need the `equals()` and `hashCode()` methods?

Comment (wernerdv) [reply server-common/src/main/java/org/apache/kafka/common/ClientIdAndBroker.java]: I added these methods for compatibility with Scala's case class with automatically defined hashcode and equals methods.
I agree that we can do without them.

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**